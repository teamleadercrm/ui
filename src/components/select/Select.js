import React, { forwardRef, PureComponent } from 'react';
import ReactSelect from 'react-select';
import ReactCreatableSelect from 'react-select/creatable';
import PropTypes from 'prop-types';
import { IconCloseBadgedSmallFilled, IconChevronDownSmallOutline } from '@teamleader/ui-icons';
import Box, { omitBoxProps, pickBoxProps } from '../box';
import Icon from '../icon';
import ValidationText from '../validationText';
import { COLOR } from '../../constants';
import theme from './theme.css';
import cx from 'classnames';
import uiUtilities from '@teamleader/ui-utilities';

const minHeightBySizeMap = {
  tiny: 24,
  small: 30,
  medium: 36,
  large: 48,
};

const DropdownIndicator = ({ selectProps: { inverse } }) => {
  return (
    <Icon
      alignItems="center"
      className={theme['dropdown-indicator']}
      color={inverse ? 'teal' : 'neutral'}
      display="flex"
      justifyContent="center"
      tint={inverse ? 'lightest' : 'darkest'}
    >
      <IconChevronDownSmallOutline />
    </Icon>
  );
};

const ClearIndicator = ({ innerProps, selectProps: { inverse } }) => {
  return (
    <Icon color={inverse ? 'teal' : 'neutral'} display="flex" tint={inverse ? 'lightest' : 'darkest'} {...innerProps}>
      <IconCloseBadgedSmallFilled />
    </Icon>
  );
};

export const selectOverlayNode = document.createElement('div');
selectOverlayNode.setAttribute('data-teamleader-ui', 'select-overlay');

const activeSelects = new Set();

class Select extends PureComponent {
  componentDidMount() {
    activeSelects.add(this);
    const isOverlayMounted = document.contains(selectOverlayNode);
    if (!isOverlayMounted) {
      document.body.appendChild(selectOverlayNode);
    }
  }

  componentWillUnmount() {
    const isLastSelect = activeSelects.size <= 1;
    if (isLastSelect) {
      document.body.removeChild(selectOverlayNode);
    }
    activeSelects.delete(this);
  }

  getClearIndicatorStyles = (base) => {
    const { inverse } = this.props;

    return {
      ...base,
      color: inverse ? COLOR.TEAL.LIGHTEST : COLOR.TEAL.DARK,
      '&:hover': {
        color: inverse ? COLOR.NEUTRAL.LIGHTEST : COLOR.TEAL.DARKEST,
      },
      cursor: 'pointer',
      svg: {
        height: '14px',
        width: '14px',
      },
    };
  };

  getControlStyles = (base, { isDisabled, isFocused }) => {
    const { error, inverse, size, success, warning, width } = this.props;

    const commonStyles = {
      ...base,
      minHeight: minHeightBySizeMap[size],
      width,
    };

    if (inverse) {
      return {
        ...commonStyles,
        backgroundColor: isDisabled ? COLOR.TEAL.DARK : COLOR.TEAL.NORMAL,
        '&:hover': {
          borderColor: !error && !warning && !success && COLOR.TEAL.LIGHT,
        },
        borderColor: error
          ? COLOR.RUBY.LIGHT
          : warning
          ? COLOR.GOLD.LIGHT
          : success
          ? COLOR.MINT.LIGHT
          : isDisabled
          ? COLOR.TEAL.DARK
          : isFocused
          ? COLOR.TEAL.LIGHT
          : COLOR.TEAL.NORMAL,
        boxShadow: error
          ? `0 0 0 1px ${COLOR.RUBY.LIGHT}`
          : warning
          ? `0 0 0 1px ${COLOR.GOLD.LIGHT}`
          : success
          ? `0 0 0 1px ${COLOR.MINT.LIGHT}`
          : isFocused
          ? `0 0 0 1px ${COLOR.TEAL.LIGHT}`
          : 'none',
      };
    }

    return {
      ...commonStyles,
      backgroundColor: isDisabled ? COLOR.NEUTRAL.NORMAL : COLOR.NEUTRAL.LIGHTEST,
      '&:hover': {
        borderColor: !error && !warning && !success && COLOR.NEUTRAL.DARKEST,
      },
      borderColor: error
        ? COLOR.RUBY.DARK
        : warning
        ? COLOR.GOLD.DARK
        : success
        ? COLOR.MINT.DARK
        : isDisabled
        ? COLOR.NEUTRAL.NORMAL
        : isFocused
        ? COLOR.NEUTRAL.DARKEST
        : COLOR.NEUTRAL.DARK,
      boxShadow: error
        ? `0 0 0 1px ${COLOR.RUBY.DARK}`
        : warning
        ? `0 0 0 1px ${COLOR.GOLD.DARK}`
        : success
        ? `0 0 0 1px ${COLOR.MINT.DARK}`
        : isFocused
        ? `0 0 0 1px ${COLOR.NEUTRAL.DARKEST}`
        : 'none',
    };
  };

  getGroupStyles = (base) => {
    const { inverse } = this.props;

    return {
      ...base,
      borderBottomColor: inverse ? COLOR.TEAL.LIGHT : COLOR.NEUTRAL.NORMAL,
      borderBottomStyle: 'solid',
      borderBottomWidth: '1px',
      '&:last-child': {
        borderWidth: 0,
      },
    };
  };

  getGroupHeadingStyles = (base) => {
    const { inverse } = this.props;

    return {
      ...base,
      color: inverse ? COLOR.NEUTRAL.LIGHTEST : COLOR.TEAL.DARKEST,
      fontSize: '12px',
      fontWeight: '700',
      letterSpacing: '0.6px',
    };
  };

  getInput = (base) => {
    return {
      ...base,
      marginLeft: '2px',
      paddingBottom: 0,
      paddingTop: 0,
    };
  };

  getMenuStyles = (base) => {
    const { menuHorizontalOffset } = this.props;
    return {
      ...base,
      marginTop: 3,
      marginBottom: 3,
      ...(menuHorizontalOffset
        ? {
            right: menuHorizontalOffset,
            position: 'absolute',
          }
        : {}),
    };
  };

  getMenuPortalStyles = (base) => {
    const { inverse, menuWidth } = this.props;

    return {
      ...base,
      ...(menuWidth && { width: menuWidth }),
      backgroundColor: inverse ? COLOR.TEAL.NORMAL : COLOR.NEUTRAL.LIGHTEST,
      fontFamily: 'Inter',
      fontSize: '14px',
      fontWeight: 400,
      fontSmoothing: 'antialiased',
      MozOsxFontSmoothing: 'grayscale',
      WebkitFontSmoothing: 'antialiased',
      zIndex: 500,
    };
  };

  getMultiValueStyles = (base) => {
    const { inverse } = this.props;

    return {
      ...base,
      borderColor: inverse ? COLOR.TEAL.DARK : COLOR.NEUTRAL.NORMAL,
      borderStyle: 'solid',
      borderWidth: '1px',
      borderRadius: '4px',
      margin: '1px',
    };
  };

  getMultiValueLabelStyles = (base) => {
    const { inverse, size } = this.props;

    return {
      ...base,
      backgroundColor: inverse ? COLOR.TEAL.DARK : COLOR.NEUTRAL.LIGHT,
      borderRadius: 0,
      color: inverse ? COLOR.NEUTRAL.LIGHTEST : COLOR.TEAL.DARKEST,
      fontFamily: 'Inter',
      fontWeight: 500,
      fontSize: size === 'tiny' || size === 'small' ? '12px' : '14px',
      lineHeight: size === 'tiny' || size === 'small' ? '1' : '18px',
      padding: size === 'tiny' ? '3px' : size === 'large' ? '9px' : '6px',
    };
  };

  getMultiValueRemoveStyles = (base) => {
    const { inverse, size } = this.props;

    return {
      ...base,
      backgroundColor: inverse ? COLOR.TEAL.DARK : COLOR.NEUTRAL.LIGHT,
      borderRadius: 0,
      color: inverse ? COLOR.NEUTRAL.LIGHTEST : COLOR.TEAL.DARKEST,
      '&:hover': {
        backgroundColor: inverse ? COLOR.TEAL.DARKEST : COLOR.NEUTRAL.NORMAL,
        color: inverse ? COLOR.NEUTRAL.LIGHTEST : COLOR.TEAL.DARKEST,
      },
      paddingLeft: size === 'tiny' ? '3px' : '6px',
      paddingRight: size === 'tiny' ? '3px' : '6px',
      transition: 'background-color .35s cubic-bezier(.4, 0, .2, 1)',
    };
  };

  getOptionStyles = (base, { isDisabled, isFocused, isSelected }) => {
    const { truncateOptionText } = this.props;
    const commonStyles = {
      ...base,
      ...(truncateOptionText
        ? {
            overflow: 'hidden',
            textOverflow: 'ellipsis',
          }
        : {
            wordBreak: 'break-word',
          }),
      padding: '8px 12px',
    };

    if (this.props.inverse) {
      return {
        ...commonStyles,
        color: isDisabled
          ? COLOR.TEAL.LIGHT
          : isSelected
          ? COLOR.NEUTRAL.LIGHTEST
          : isFocused
          ? COLOR.TEAL.DARK
          : COLOR.NEUTRAL.LIGHTEST,
        backgroundColor: isSelected ? COLOR.TEAL.DARK : isFocused ? COLOR.TEAL.LIGHT : COLOR.TEAL.NORMAL,
        '&:active': {
          backgroundColor: isDisabled ? COLOR.TEAL.NORMAL : COLOR.TEAL.DARK,
          color: isDisabled ? COLOR.TEAL.LIGHT : COLOR.NEUTRAL.LIGHTEST,
        },
      };
    }

    return {
      ...commonStyles,
      color: isDisabled ? COLOR.NEUTRAL.DARK : COLOR.TEAL.DARK,
      backgroundColor: isSelected ? COLOR.AQUA.LIGHTEST : isFocused ? COLOR.NEUTRAL.LIGHT : COLOR.NEUTRAL.LIGHTEST,
      '&:active': {
        backgroundColor: isDisabled ? COLOR.NEUTRAL.LIGHTEST : COLOR.AQUA.LIGHTEST,
        color: isDisabled ? COLOR.NEUTRAL.DARK : COLOR.TEAL.DARK,
      },
    };
  };

  getPlaceholderStyles = (base, { isDisabled }) => {
    const { inverse } = this.props;

    const commonStyles = {
      ...base,
      marginLeft: '2px',
      marginRight: '2px',
      whiteSpace: 'nowrap',
    };

    if (inverse) {
      return {
        ...commonStyles,
        color: isDisabled ? COLOR.TEAL.NORMAL : COLOR.TEAL.LIGHT,
      };
    }

    return {
      ...commonStyles,
      color: COLOR.NEUTRAL.DARKEST,
    };
  };

  getSingleValueStyles = (base) => ({
    ...base,
    color: this.props.inverse ? COLOR.NEUTRAL.LIGHTEST : COLOR.TEAL.DARKEST,
  });

  getValueContainerStyles = (base, { isMulti, hasValue }) => {
    const { size } = this.props;

    return {
      ...base,
      minHeight: minHeightBySizeMap[size] - 2,
      lineHeight: '18px',
      padding: isMulti && hasValue && size !== 'large' ? 0 : '0 4px',
    };
  };

  getStyles = () => ({
    clearIndicator: this.getClearIndicatorStyles,
    control: this.getControlStyles,
    group: this.getGroupStyles,
    groupHeading: this.getGroupHeadingStyles,
    input: this.getInput,
    menu: this.getMenuStyles,
    menuPortal: this.getMenuPortalStyles,
    multiValue: this.getMultiValueStyles,
    multiValueLabel: this.getMultiValueLabelStyles,
    multiValueRemove: this.getMultiValueRemoveStyles,
    option: this.getOptionStyles,
    placeholder: this.getPlaceholderStyles,
    singleValue: this.getSingleValueStyles,
    valueContainer: this.getValueContainerStyles,
  });

  render() {
    const {
      components,
      noOptionsMessage = 'No options',
      creatable,
      error,
      inverse,
      helpText,
      size,
      success,
      warning,
      forwardedRef,
      ...otherProps
    } = this.props;

    const boxProps = pickBoxProps(otherProps);
    const restProps = omitBoxProps(otherProps);

    const wrapperClassnames = cx(theme[`is-${size}`], {
      [theme['has-error']]: error,
      [theme['is-inverse']]: inverse,
    });

    const Element = creatable ? ReactCreatableSelect : ReactSelect;

    return (
      <Box className={wrapperClassnames} {...boxProps}>
        <Element
          ref={forwardedRef}
          className={cx(uiUtilities['reset-font-smoothing'], theme['select'])}
          components={{
            ClearIndicator,
            DropdownIndicator,
            IndicatorSeparator: null,
            ...components,
          }}
          hideSelectedOptions={false}
          menuPlacement="auto"
          menuPortalTarget={selectOverlayNode}
          menuShouldBlockScroll
          styles={this.getStyles()}
          size={size}
          noOptionsMessage={() => noOptionsMessage}
          {...restProps}
        />
        <ValidationText error={error} help={helpText} inverse={inverse} success={success} warning={warning} />
      </Box>
    );
  }
}

Select.propTypes = {
  /** Override default components with your own. Pass an object with correct the key and its replacing component */
  components: PropTypes.object,
  /** If true, it's possible to create a new option that is not in the list. */
  creatable: PropTypes.bool,
  /** The text string/element to use as error message below the input. */
  error: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  /** The text string to use as help text below the input. */
  helpText: PropTypes.string,
  /** Boolean indicating whether the select should render as inverse. */
  inverse: PropTypes.bool,
  /** A custom width for the menu dropdown */
  menuWidth: PropTypes.string,
  /** A custom horizontal offset for the menu dropdown, useful when also using a custom menuWidth  */
  menuHorizontalOffset: PropTypes.string,
  /** Boolean indicating whether the select option text should render on one single line. */
  truncateOptionText: PropTypes.bool,
  /** Size of the input element. */
  size: PropTypes.oneOf(['tiny', 'small', 'medium', 'large']),
  /** The text string/element to use as success message below the input. */
  success: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  /** Selected option value(s) */
  value: PropTypes.oneOfType([PropTypes.object, PropTypes.array, PropTypes.func]),
  /** The text to use as warning message below the input. */
  warning: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  /** A custom width for the input field */
  width: PropTypes.string,
  /** Message shown where are no options */
  noOptionsMessage: PropTypes.string,
};

Select.defaultProps = {
  creatable: false,
  inverse: false,
  size: 'medium',
  width: '100%',
};

/** @type {React.ComponentType<any>} */
const ForwardedSelect = forwardRef((props, ref) => <Select {...props} forwardedRef={ref} />);

ForwardedSelect.displayName = 'Select';

export default ForwardedSelect;
