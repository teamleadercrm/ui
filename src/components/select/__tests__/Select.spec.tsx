import React from 'react';
import { render } from '@testing-library/react';
import Select from '../Select';

describe('Component - Select', () => {
  it('renders', async () => {
    const { asFragment } = render(
      <Select
        options={[
          { label: 'Michael Scoott', value: '1' },
          { label: 'Jim Halpert', value: '2' },
        ]}
        onChange={() => {}}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
