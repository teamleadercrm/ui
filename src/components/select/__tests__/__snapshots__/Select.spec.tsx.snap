// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Select does not render a clear button when no value is provided 1`] = `
<DocumentFragment>
  <div
    class="box is-medium"
    data-teamleader-ui="select"
  >
    <div
      class="reset-font-smoothing select css-b62m3t-container"
    >
      <span
        class="css-1f43avz-a11yText-A11yText"
        id="react-select-12-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="css-1f43avz-a11yText-A11yText"
      />
      <div
        class=" css-8v26ep-control"
      >
        <div
          class=" css-wbqiiw-ValueContainer"
        >
          <div
            class=" css-crj073-placeholder"
            id="react-select-12-placeholder"
          >
            Select...
          </div>
          <div
            class=" css-19m4yby-Input"
            data-value=""
          >
            <input
              aria-autocomplete="list"
              aria-describedby="react-select-12-placeholder"
              aria-expanded="false"
              aria-haspopup="true"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              class=""
              id="react-select-12-input"
              role="combobox"
              spellcheck="false"
              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <span
            class="box align-items-center display-flex justify-content-center neutral darkest dropdown-indicator"
            data-teamleader-ui="icon"
          >
            <svg
              fill="currentColor"
              height="14"
              preserveAspectRatio="xMidYMid meet"
              style="opacity: 0.84; vertical-align: middle;"
              viewBox="0 0 14 14"
              width="14"
            >
              <path
                clip-rule="evenodd"
                d="M3.465 6.087L7 9.623l3.536-3.536A1 1 0 009.12 4.673L7 6.794 4.878 4.673a1 1 0 10-1.414 1.414z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Component - Select renders 1`] = `
<DocumentFragment>
  <div
    class="box is-medium"
    data-teamleader-ui="select"
  >
    <div
      class="reset-font-smoothing select css-b62m3t-container"
    >
      <span
        class="css-1f43avz-a11yText-A11yText"
        id="react-select-2-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="css-1f43avz-a11yText-A11yText"
      />
      <div
        class=" css-8v26ep-control"
      >
        <div
          class=" css-wbqiiw-ValueContainer"
        >
          <div
            class=" css-crj073-placeholder"
            id="react-select-2-placeholder"
          >
            Select...
          </div>
          <div
            class=" css-19m4yby-Input"
            data-value=""
          >
            <input
              aria-autocomplete="list"
              aria-describedby="react-select-2-placeholder"
              aria-expanded="false"
              aria-haspopup="true"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              class=""
              id="react-select-2-input"
              role="combobox"
              spellcheck="false"
              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <span
            class="box align-items-center display-flex justify-content-center neutral darkest dropdown-indicator"
            data-teamleader-ui="icon"
          >
            <svg
              fill="currentColor"
              height="14"
              preserveAspectRatio="xMidYMid meet"
              style="opacity: 0.84; vertical-align: middle;"
              viewBox="0 0 14 14"
              width="14"
            >
              <path
                clip-rule="evenodd"
                d="M3.465 6.087L7 9.623l3.536-3.536A1 1 0 009.12 4.673L7 6.794 4.878 4.673a1 1 0 10-1.414 1.414z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Component - Select renders a clear button when a value is provided 1`] = `
<DocumentFragment>
  <div
    class="box is-medium"
    data-teamleader-ui="select"
  >
    <div
      class="reset-font-smoothing select css-b62m3t-container"
    >
      <span
        class="css-1f43avz-a11yText-A11yText"
        id="react-select-11-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="css-1f43avz-a11yText-A11yText"
      />
      <div
        class=" css-8v26ep-control"
      >
        <div
          class=" css-wbqiiw-ValueContainer"
        >
          <div
            class=" css-1i5bhiv-singleValue"
          >
            Michael Scoott
          </div>
          <div
            class=" css-19m4yby-Input"
            data-value=""
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-haspopup="true"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              class=""
              id="react-select-11-input"
              role="combobox"
              spellcheck="false"
              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <span
            aria-hidden="true"
            class="box align-items-center display-flex neutral darkest clear-indicator"
            data-teamleader-ui="icon"
          >
            <svg
              fill="currentColor"
              height="14"
              preserveAspectRatio="xMidYMid meet"
              style="opacity: 0.84; vertical-align: middle;"
              viewBox="0 0 14 14"
              width="14"
            >
              <path
                clip-rule="evenodd"
                d="M7 13A6 6 0 117 1a6 6 0 010 12zm0-4.586l1.414 1.414a1 1 0 101.414-1.414L8.414 7l1.414-1.414a1 1 0 10-1.414-1.414L7 5.586 5.586 4.172a1 1 0 00-1.414 1.414L5.586 7 4.172 8.414a1 1 0 001.414 1.414L7 8.414z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="box align-items-center display-flex justify-content-center neutral darkest dropdown-indicator"
            data-teamleader-ui="icon"
          >
            <svg
              fill="currentColor"
              height="14"
              preserveAspectRatio="xMidYMid meet"
              style="opacity: 0.84; vertical-align: middle;"
              viewBox="0 0 14 14"
              width="14"
            >
              <path
                clip-rule="evenodd"
                d="M3.465 6.087L7 9.623l3.536-3.536A1 1 0 009.12 4.673L7 6.794 4.878 4.673a1 1 0 10-1.414 1.414z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Component - Select renders a help text 1`] = `
<DocumentFragment>
  <div
    class="box is-medium"
    data-teamleader-ui="select"
  >
    <div
      class="reset-font-smoothing select css-b62m3t-container"
    >
      <span
        class="css-1f43avz-a11yText-A11yText"
        id="react-select-10-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="css-1f43avz-a11yText-A11yText"
      />
      <div
        class=" css-8v26ep-control"
      >
        <div
          class=" css-wbqiiw-ValueContainer"
        >
          <div
            class=" css-crj073-placeholder"
            id="react-select-10-placeholder"
          >
            Select...
          </div>
          <div
            class=" css-19m4yby-Input"
            data-value=""
          >
            <input
              aria-autocomplete="list"
              aria-describedby="react-select-10-placeholder"
              aria-expanded="false"
              aria-haspopup="true"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              class=""
              id="react-select-10-input"
              role="combobox"
              spellcheck="false"
              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <span
            class="box align-items-center display-flex justify-content-center neutral darkest dropdown-indicator"
            data-teamleader-ui="icon"
          >
            <svg
              fill="currentColor"
              height="14"
              preserveAspectRatio="xMidYMid meet"
              style="opacity: 0.84; vertical-align: middle;"
              viewBox="0 0 14 14"
              width="14"
            >
              <path
                clip-rule="evenodd"
                d="M3.465 6.087L7 9.623l3.536-3.536A1 1 0 009.12 4.673L7 6.794 4.878 4.673a1 1 0 10-1.414 1.414z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <p
      class="box margin-top-1 text text-small darkest neutral"
      data-teamleader-ui="help-text"
    >
      Fake help text!
    </p>
  </div>
</DocumentFragment>
`;

exports[`Component - Select renders a success message 1`] = `
<DocumentFragment>
  <div
    class="box is-medium"
    data-teamleader-ui="select"
  >
    <div
      class="reset-font-smoothing select css-b62m3t-container"
    >
      <span
        class="css-1f43avz-a11yText-A11yText"
        id="react-select-8-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="css-1f43avz-a11yText-A11yText"
      />
      <div
        class=" css-1ecujp2-control"
      >
        <div
          class=" css-wbqiiw-ValueContainer"
        >
          <div
            class=" css-crj073-placeholder"
            id="react-select-8-placeholder"
          >
            Select...
          </div>
          <div
            class=" css-19m4yby-Input"
            data-value=""
          >
            <input
              aria-autocomplete="list"
              aria-describedby="react-select-8-placeholder"
              aria-expanded="false"
              aria-haspopup="true"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              class=""
              id="react-select-8-input"
              role="combobox"
              spellcheck="false"
              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <span
            class="box align-items-center display-flex justify-content-center neutral darkest dropdown-indicator"
            data-teamleader-ui="icon"
          >
            <svg
              fill="currentColor"
              height="14"
              preserveAspectRatio="xMidYMid meet"
              style="opacity: 0.84; vertical-align: middle;"
              viewBox="0 0 14 14"
              width="14"
            >
              <path
                clip-rule="evenodd"
                d="M3.465 6.087L7 9.623l3.536-3.536A1 1 0 009.12 4.673L7 6.794 4.878 4.673a1 1 0 10-1.414 1.414z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <p
      class="box margin-top-1 text text-small dark mint"
      data-teamleader-ui="success-text"
    >
      <span>
        Fake success message!
      </span>
    </p>
  </div>
</DocumentFragment>
`;

exports[`Component - Select renders a warning message 1`] = `
<DocumentFragment>
  <div
    class="box is-medium"
    data-teamleader-ui="select"
  >
    <div
      class="reset-font-smoothing select css-b62m3t-container"
    >
      <span
        class="css-1f43avz-a11yText-A11yText"
        id="react-select-9-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="css-1f43avz-a11yText-A11yText"
      />
      <div
        class=" css-1qn4nre-control"
      >
        <div
          class=" css-wbqiiw-ValueContainer"
        >
          <div
            class=" css-crj073-placeholder"
            id="react-select-9-placeholder"
          >
            Select...
          </div>
          <div
            class=" css-19m4yby-Input"
            data-value=""
          >
            <input
              aria-autocomplete="list"
              aria-describedby="react-select-9-placeholder"
              aria-expanded="false"
              aria-haspopup="true"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              class=""
              id="react-select-9-input"
              role="combobox"
              spellcheck="false"
              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <span
            class="box align-items-center display-flex justify-content-center neutral darkest dropdown-indicator"
            data-teamleader-ui="icon"
          >
            <svg
              fill="currentColor"
              height="14"
              preserveAspectRatio="xMidYMid meet"
              style="opacity: 0.84; vertical-align: middle;"
              viewBox="0 0 14 14"
              width="14"
            >
              <path
                clip-rule="evenodd"
                d="M3.465 6.087L7 9.623l3.536-3.536A1 1 0 009.12 4.673L7 6.794 4.878 4.673a1 1 0 10-1.414 1.414z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <p
      class="box margin-top-1 text text-small dark gold"
      data-teamleader-ui="warning-text"
    >
      <span>
        Fake warning message!
      </span>
    </p>
  </div>
</DocumentFragment>
`;

exports[`Component - Select renders an error message 1`] = `
<DocumentFragment>
  <div
    class="box is-medium has-error"
    data-teamleader-ui="select"
  >
    <div
      class="reset-font-smoothing select css-b62m3t-container"
    >
      <span
        class="css-1f43avz-a11yText-A11yText"
        id="react-select-7-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="css-1f43avz-a11yText-A11yText"
      />
      <div
        class=" css-153luwj-control"
      >
        <div
          class=" css-wbqiiw-ValueContainer"
        >
          <div
            class=" css-crj073-placeholder"
            id="react-select-7-placeholder"
          >
            Select...
          </div>
          <div
            class=" css-19m4yby-Input"
            data-value=""
          >
            <input
              aria-autocomplete="list"
              aria-describedby="react-select-7-placeholder"
              aria-expanded="false"
              aria-haspopup="true"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              class=""
              id="react-select-7-input"
              role="combobox"
              spellcheck="false"
              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <span
            class="box align-items-center display-flex justify-content-center neutral darkest dropdown-indicator"
            data-teamleader-ui="icon"
          >
            <svg
              fill="currentColor"
              height="14"
              preserveAspectRatio="xMidYMid meet"
              style="opacity: 0.84; vertical-align: middle;"
              viewBox="0 0 14 14"
              width="14"
            >
              <path
                clip-rule="evenodd"
                d="M3.465 6.087L7 9.623l3.536-3.536A1 1 0 009.12 4.673L7 6.794 4.878 4.673a1 1 0 10-1.414 1.414z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <p
      class="box margin-top-1 text text-small dark ruby"
      data-teamleader-ui="error-text"
    >
      <span>
        Fake error message!
      </span>
    </p>
  </div>
</DocumentFragment>
`;
